src/shared/components/lock-screen/LockScreen.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Lottie from 'lottie-react-web';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import styles from './LockScreen.css';
+import animation from '../../media/animations/unlock-animation.json';
+import background from '../../media/images/background-pattern.svg';
+
+const LOCK_TYPE = 'passphrase';
+
+class LockScreen extends Component {
+    passwordInputRef = React.createRef();
+    animationOptions = {
+        animationData: animation,
+        loop: false,
+    };
+
+    state = {
+        passwordLength: 0,
+        feedback: 'none',
+        startLogoAnimation: false,
+    };
+
+    componentDidMount() {
+        const input = this.passwordInputRef.current;
+
+        input.focus();
+        input.addEventListener('keydown', this.handleKeyboardInput);
+        input.addEventListener('blur', () => input.focus());
+
+        this.mounted = false;
+        this.forceUpdate();
+    }
+
+    render() {
+        const { passwordLength, feedback, startLogoAnimation } = this.state;
+        const { unmounting } = this.props;
+        const passwordClassName = classNames(
+            styles.passwordDot,
+            feedback === 'loading' && styles.loading,
+            feedback === 'error' && styles.error
+        );
+
+        const animationDelay = feedback === 'loading' ? -0.1 : 0;
+
+        return (
+            <div className={ styles.lockscreen } onClick={ this.handleMouseClick }>
+                <div className={ styles.background } dangerouslySetInnerHTML={ { __html: background } } />
+                <CSSTransition classNames={ styles.lockscreenContent } in={ !unmounting } appear component={ null } timeout={ 1500 }>
```

Comment:
My argument behind using `:global` is that defining all the individual classnames applied by `CSSTransition` is rather verbose. Unless there is a performance drawback in using `:global`, I say it could stay as it is. Any thoughts on this @satazor ?